helper.go

```diff
@@ -509,6 +510,9 @@ func parseOpaque(logger Logger, data []byte, retVal *variable) error {
 			if length > len(data) {
 				return fmt.Errorf("not enough data for OpaqueFloat %x (data %d length %d)", data, len(data), length)
 			}
+			if cursor > length {
+				return fmt.Errorf("invalid cursor position for OpaqueFloat %x (data %d length %d cursor %d)", data, len(data), length, cursor)
+			}
```

Comment:
If the cursor is greater than length then a panic would be observed on line 513 (517) below.